cmd/kube-proxy/app/server.go

```diff
@@ -1084,10 +1085,15 @@ func detectNodeIPs(ctx context.Context, rawNodeIPs []net.IP, bindAddress string)
 	return primaryFamily, nodeIPs
 }
 
-// getNodeIP returns IPs for the node with the provided name.  If
+// getNodeIP returns IPs for the node with the provided name. If
 // required, it will wait for the node to be created.
 func getNodeIPs(ctx context.Context, client clientset.Interface, name string) []net.IP {
 	logger := klog.FromContext(ctx)
+
+	if err := waitUntilNodeCache(ctx, client, name); err != nil {
```

Comment:
doing a watch first to get the node and after that run a Get() directly to the apiserver can be optimized to just get the node IPs directly from the watch https://github.com/kubernetes/kubernetes/pull/124670/files#r1629187355